### PR TITLE
osv-scanner used as a library with more reliable filtering

### PR DESCRIPTION
**Pull request by briangann**
This adds a revamped osv-scanner that calls the latest version (1.2.0) as a library instead of spawning a process.

Filtering has also been modified to auto-detect issues attributed to `@grafana/*` packages, without enumerating any of the details.  This is effectively a generic/automated allow list no matter which version of the grafana packages are used.

The old spawned method has been removed

- [x] Removes external dependency spawning osv-scanner
- [x] Update documentation
- [x] Remove osv-scanner binary from build process
- [x] Migrate all tests from previous method

### CODE REVIEW

File: pkg/analysis/passes/osvscanner-internal/osv_scanner_internal.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+	// check for the different files we check for and run scanner on each
+	// if none are detected report OK
+	scanningPerformed := false
+	for _, scannerType := range scannerTypes {
+		lockFile := filepath.Join(sourceCodeDir, scannerType)
+		if _, err := os.Stat(lockFile); err == nil {
+			// perform scan
+			scanningPerformed = true
+			data, _ := scanInternal(lockFile)
```
Comment (briangann) [inline pkg/analysis/passes/osvscanner-internal/osv_scanner_internal.go:64]: is there an error to detect here?

File: pkg/analysis/passes/osvscanner-internal/lockfile/search.go
Comment (briangann) [inline pkg/analysis/passes/osvscanner-internal/lockfile/search.go:1]: rename file to yarnwhy

File: pkg/analysis/passes/osvscanner-internal/osv_scanner_internal.go
Comment (briangann) [inline pkg/analysis/passes/osvscanner-internal/osv_scanner_internal.go:1]: rename to osv-scanner-library vs internal

**Review by academo — APPROVED** [review]
Thanks for the work you put on this one @briangann !

We should follow up this with a `packageJson.lock` scanning and protection since there are plugins using npm or pnpm and we are about to introduce support for them here https://github.com/grafana/plugin-tools/pull/226

**Pull request merged**